objective.py

```diff
@@ -1,26 +1,73 @@
-import numpy as np
 from benchopt import BaseObjective
+import numpy as np
+from scipy import optimize
 
 
 class Objective(BaseObjective):
     name = "Ordinary Least Squares"
 
-    parameters = {'reg': np.linspace(0.1, 0.9, 3).round(1)}
-
-    def __init__(self, reg=0.5):
-        self.reg = reg  # 0<reg<1
+    parameters = {'reg': [0.5],
+                  'delta': [0.9],
+                  'data_fit': ['quad', 'huber']
+                  # 'reg': np.linspace(0.1, 0.9, 3).round(1)
+                  }
 
     def set_data(self, A, y):
         self.A = A
         self.y = y
         S = np.sum(self.A, axis=1)
-        c = S.dot(self.y)/(S.dot(S))*np.eye(len(self.y), dtype=int)
-        reg_max = np.max(abs((self.A).T.dot(self.A.dot(c) - self.y)))
-        self.reg = self.reg*reg_max
+        self.c = self.set_c(S, self.delta)
+        self.reg = self.reg*self.set_reg_max(self.c)
 
     def compute(self, u):
         R = self.y - self.A @ u
-        return .5 * R @ R + self.reg*(abs(np.diff(u)).sum())
+        if self.data_fit == 'quad':
+            return .5 * R @ R + self.reg*(abs(np.diff(u)).sum())
+        else:
+            return self.huber(R, self.delta) + self.reg*(abs(np.diff(u)).sum())
 
     def to_dict(self):
-        return dict(A=self.A, reg=self.reg, y=self.y)
+        return dict(A=self.A, reg=self.reg, y=self.y, c=self.c,
+                    delta=self.delta, data_fit=self.data_fit)
+
+    def huber(self, R, delta):
+        norm_1 = np.abs(R)
+        loss = np.where(norm_1 < delta,
+                        0.5 * norm_1**2,
+                        delta * norm_1 - 0.5 * delta**2)
+        return np.sum(loss)
+
+    def set_c(self, S, delta):
```

Comment:
This function does not set anything so it should be called `get_c` or `compute_c`.